xbmc/video/VideoItemArtworkHandler.cpp

```diff
@@ -444,6 +444,104 @@ void CVideoItemArtworkMovieSetHandler::AddItemPathToFileBrowserSources(
       "* " + g_localizeStrings.Get(20226) /* Movie set information folder */);
 }
 
+//-------------------------------------------------------------------------------------------------
+// CVideoItemArtworkFanartHandler (Handler for all media types, to manage fanart art type)
+//-------------------------------------------------------------------------------------------------
+
+class CVideoItemArtworkFanartHandler : public CVideoItemArtworkHandler
+{
+public:
+  explicit CVideoItemArtworkFanartHandler(const std::shared_ptr<CFileItem>& item,
+                                          const std::string& artType)
+    : CVideoItemArtworkHandler(item, artType)
+  {
+    // Ensure the fanart is unpacked
+    m_item->GetVideoInfoTag()->m_fanart.Unpack();
+  }
+
+  std::string GetCurrentArt() override;
+  std::vector<std::string> GetRemoteArt() override;
+  std::string GetLocalArt() override;
+
+  std::string GetDefaultIcon() override { return "DefaultPicture.png"; }
+  bool SupportsFlippedArt() override { return true; }
+
+  std::string UpdateEmbeddedArt(const std::string& art) override;
+  std::string UpdateRemoteArt(const std::vector<std::string>& art, int index) override;
+};
+
+std::string CVideoItemArtworkFanartHandler::GetCurrentArt()
+{
+  return m_item->GetArt("fanart");
+}
+
+std::vector<std::string> CVideoItemArtworkFanartHandler::GetRemoteArt()
+{
+  std::vector<std::string> remoteArt;
+  const CVideoInfoTag* videoTag = m_item->GetVideoInfoTag();
+  for (unsigned int i = 0; i < videoTag->m_fanart.GetNumFanarts(); ++i)
+  {
+    const std::string thumb = videoTag->m_fanart.GetPreviewURL(i);
+    if (URIUtils::IsProtocol(thumb, "image"))
+      continue;
+
+    remoteArt.emplace_back(CTextureUtils::GetWrappedThumbURL(thumb));
+  }
+  return remoteArt;
+}
+
+std::string CVideoItemArtworkFanartHandler::GetLocalArt()
+{
+  return m_item->GetLocalFanart();
+}
+
+std::string CVideoItemArtworkFanartHandler::UpdateEmbeddedArt(const std::string& art)
+{
+  CVideoDatabase videodb;
+  if (!videodb.Open())
+  {
+    CLog::LogF(LOGERROR, "Cannot open video database!");
+    return art;
+  }
+
+  CVideoInfoTag* videoTag = m_item->GetVideoInfoTag();
+  const int current = videoTag->m_fanart.GetNumFanarts();
+  int found = -1;
```

Comment:
found implicitly reads as a boolean. Maybe rename it to index or similar?

Reply:
Good point, took over the name from existing implementation, will change this.